### PR TITLE
Test plan for  QUARKUS-1026 Continuous Testing in Dev Mode

### PR DESCRIPTION
**Pull request by Sgitario**

### CODE REVIEW

File: QUARKUS-1026.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- Run affected tests after new changes when continuous testing is enabled
+- Dev services should be deployed/undeployed for tests (if required) when continuous testing is enabled
+ - Kafka Dev services
+ - Database Dev Services
+ - Keycloak Dev Services
+- Cover gRPC testing when continuous testing is enabled
+ - gRPC is an special use case as it requires another server listening to a local port
+
+### Impact on testsuites and testing automation:
+- Enhance existing test suite / test framework
```
Comment (rsvoboda) [inline QUARKUS-1026.md:20]: What's the time estimation for this effort ?

File: QUARKUS-1026.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- Ensure this coverage works fine on DEV mode
+- Ensure this coverage works fine on Unix or Windows environments
```
Comment (rsvoboda) [inline QUARKUS-1026.md:22]: Docker is prerequisite and we do not have that on Windows machines in PSI nor GHActions.

File: QUARKUS-1026.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## Scope of the testing
+- Enable/Disable Continuous Testing on a running Quarkus application
+- Cover DEV UI functionality about continuous testing
+- Detect new tests when continuous testing is enabled
+- Run affected tests after new changes when continuous testing is enabled
+- Dev services should be deployed/undeployed for tests (if required) when continuous testing is enabled
+ - Kafka Dev services
+ - Database Dev Services
+ - Keycloak Dev Services
```
Comment (rsvoboda) [inline QUARKUS-1026.md:15]: I would mark Keycloak as optional

**Comment by rsvoboda** [comment]
Kyrylo is backup for this epic, adding him as reviewer

File: QUARKUS-1026.md (hunk 1 above)
Comment (Sgitario) [reply QUARKUS-1026.md:20]: Everything should have been done already.... so I don't think we need to spend more time to support continuous testing.

File: QUARKUS-1026.md (hunk 2 above)
Comment (Sgitario) [reply QUARKUS-1026.md:22]: Do you want me to add this note here or descope Windows coverage?

File: QUARKUS-1026.md (hunk 3 above)
Comment (Sgitario) [reply QUARKUS-1026.md:15]: okis

File: QUARKUS-1026.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Test plan covers Continuous Testing in Dev Mode functionality: https://quarkus.io/guides/continuous-testing
+
+## Scope of the testing
+- Enable/Disable Continuous Testing on a running Quarkus application
+- Cover DEV UI functionality about continuous testing
+- Detect new tests when continuous testing is enabled
+- Run affected tests after new changes when continuous testing is enabled
+- Dev services should be deployed/undeployed for tests (if required) when continuous testing is enabled
+ - Kafka Dev services
```
Comment (kshpak) [inline QUARKUS-1026.md:13]: To consider:
AMQP Dev services https://quarkus.io/guides/amqp-dev-services

File: QUARKUS-1026.md (hunk 2 above)
Comment (Sgitario) [reply QUARKUS-1026.md:22]: I've added a note.

File: QUARKUS-1026.md (hunk 4 above)
Comment (Sgitario) [reply QUARKUS-1026.md:13]: Added

**Review by kshpak — APPROVED** [review]

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**